xarray/conventions.py

```diff
@@ -279,8 +271,6 @@ def encode_cf_datetime(dates, units=None, calendar=None):
 
     if units is None:
         units = infer_datetime_units(dates)
-    else:
```

Comment:
my approach would be not to change the user-specified units string, as this can be surprising. But there might have been a reason for this...